src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs

```diff
@@ -281,7 +281,9 @@ public override IEnumerable<IEdmVocabularyAnnotation> FindDeclaredVocabularyAnno
                 {
                     foreach (CsdlAnnotation annotation in annotations.Annotations.Annotations)
                     {
-                        result.Add(this.WrapVocabularyAnnotation(annotation, annotations.Context, null, annotations, annotations.Annotations.Qualifier));
+                        IEdmVocabularyAnnotation vocabAnnotation = this.WrapVocabularyAnnotation(annotation, annotations.Context, null, annotations, annotations.Annotations.Qualifier);
+                        vocabAnnotation.SetSerializationLocation(this, EdmVocabularyAnnotationSerializationLocation.OutOfLine);
+                        result.Add(vocabAnnotation);

```

Comment:
The change here is straightforward to call "SetSerializationLocation" to set "OutOfLine" for the annotation.

Reply:
Help me understand why we're doing this? Why are we changing all of the annotations to OutOflLine?  Does this mean that, if I read in some inline annotations, then write them back out, they will all be out of line?

---
In reply to: [221688408](https://github.com/OData/odata.net/pull/1263#discussion_r221688408) [](ancestors = 221688408)

Reply:
Ok. 
The function `FindDeclaredVocabularyAnnotations(element)` is to return all annotations related to the given element, include all "**InLine**" and "**OutOfLine**" annotations. 
For the "**InLine**" annotation, we have the codes to set `SerializationLocation` as `InLine`, see codes [here](https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs#L507-L509)

But, for the **OutOfLine** annotation, we miss to set `SerializationLocation` for them. If we call "GetSerializationLocation", it will give you `null`. So, I changed the codes to set location as "**OutOfLine**" for them, same as for "**InLine**" annotation.

That's straightforward because the annotations are from the "**outOfLineAnnotations**" collection, see the following `if` clause.

```C#
 if (fullName != null && this.outOfLineAnnotations.TryGetValue(fullName, out elementAnnotations))
 {
         List<IEdmVocabularyAnnotation> result = new List<IEdmVocabularyAnnotation>();
         foreach (CsdlSemanticsAnnotations annotations in elementAnnotations)
         {
               foreach (CsdlAnnotation annotation in annotations.Annotations.Annotations)
                {
                     IEdmVocabularyAnnotation vocabAnnotation = this.WrapVocabularyAnnotation(annotation, annotations.Context, null, annotations, annotations.Annotations.Qualifier);
                     vocabAnnotation.SetSerializationLocation(this, EdmVocabularyAnnotationSerializationLocation.OutOfLine);
                     result.Add(vocabAnnotation);
                }
           }

         return inlineAnnotations.Concat(result);
  }

Reply:
I see; thanks.  Part of what was confusing me was the fact that we are concatenating the annotations to an enumerable named "inlineAnnotations"; I thought that we were setting all of the annotations to out of line, but now see that we were concatenating the outOfLine annotations to the inlineAnnotations and returning that concatenated list...